### PR TITLE
chore: renovatebot rebase when behind

### PR DESCRIPTION
**Pull request by noahdietz**
Instructs renovatebot to rebase when the branch is behind
Changes `assignees` to `reviewers`
Adds label `automerge` on PR creation

### CODE REVIEW

File: renovate.json [hunk 1]
```diff
@@ -11,7 +11,9 @@
     ],
     "postUpdateOptions": ["gomodTidy"]
   },
-  "assignees": [
+  "reviewers": [
     "noahdietz"
-  ]
+  ],
+  "rebaseWhen": "behind-base-branch",
+  "labels": ["automerge"]
```
Comment (tbpg) [inline renovate.json:18]: I think this will cause problems if the PR isn't reviewed within MoG's merge window. Not sure.

Comment (noahdietz) [reply renovate.json:18]: Spoke offline with Sofia and she said it shouldn't be a problem. Readding the label makes everything OK if I don't handle within the 6hr window

**Comment by noahdietz** [comment]
@tbpg I added a package rule to schedule genproto updates. Can you provide a sanity check?

**Comment by noahdietz** [comment]
Here is what renovate said:

```
There is an error with this repository's Renovate configuration that needs to be fixed. As a precaution, Renovate will stop PRs until it is resolved.

Error type: undefined
```

😆

**Pull request merged**